frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/index.controller.ts

```diff
@@ -0,0 +1,422 @@
+  }
+
+  private handleVisibilityChange() {
+    if (document.hidden) {
+      this.stopPolling();
+    } else {
+      void this.updateActivitiesList();
+      this.startPolling();
+    }
+  }
+
+  private startPolling() {
+    if (this.intervallId) window.clearInterval(this.intervallId);
+    this.intervallId = this.pollForUpdates();
+  }
+
+  private stopPolling() {
+    window.clearInterval(this.intervallId);
+  }
+
+  private pollForUpdates() {
+    return window.setInterval(() => this.updateActivitiesList(), this.pollingIntervalInMsValue);
+  }
+
+  handleWorkPackageUpdate(_event:Event):void {
+    setTimeout(() => this.updateActivitiesList(), 2000);
+  }
+
+  async updateActivitiesList() {
+    const journalsContainerAtBottom = this.isJournalsContainerScrolledToBottom(this.journalsContainerTarget);
+    const url = new URL(this.updateStreamsUrlValue);
+    url.searchParams.append('last_update_timestamp', this.lastUpdateTimestamp);
+    url.searchParams.append('filter', this.filterValue);
+
+    const response = await this.fetchWithCSRF(url, 'GET');
+
+    if (response.ok) {
+      const text = await response.text();
+      Turbo.renderStreamMessage(text);
+      this.setLastUpdateTimestamp();
+      setTimeout(() => {
+        if (this.sortingValue === 'asc' && journalsContainerAtBottom) {
+          // scroll to (new) bottom if sorting is ascending and journals container was already at bottom before a new activity was added
+          if (this.isMobile()) {
+            this.scrollInputContainerIntoView(300);
+          } else {
+            this.scrollJournalContainer(this.journalsContainerTarget, true);
+          }
+        }
+      }, 100);
+    }
+  }
+
+  private rescueEditorContent() {
+    const ckEditorInstance = this.getCkEditorInstance();
+    if (ckEditorInstance) {
+      const data = ckEditorInstance.getData({ trim: false });
+      if (data.length > 0) {
+        localStorage.setItem(this.localStorageKey, data);
+      }
+    }
+  }
+
+  private populateRescuedEditorContent() {
+    const rescuedEditorContent = localStorage.getItem(this.localStorageKey);
+    if (rescuedEditorContent) {
+      this.openEditorWithInitialData(rescuedEditorContent);
+      localStorage.removeItem(this.localStorageKey);
+    }
+  }
+
+  private handleInitialScroll() {
+    if (window.location.hash.includes('#activity-')) {
+      this.scrollToActivity();
+    } else if (this.sortingValue === 'asc') {
+      this.scrollToBottom();
+    }
+  }
+
+  private scrollToActivity() {
+    const activityId = window.location.hash.replace('#activity-', '');
+    const activityElement = document.getElementById(`activity-${activityId}`);
+    activityElement?.scrollIntoView({ behavior: 'smooth' });
+  }
+
+  private scrollToBottom() {
+    const scrollableContainer = jQuery(this.element).scrollParent()[0];
+    if (scrollableContainer) {
+      setTimeout(() => {
+        scrollableContainer.scrollTop = scrollableContainer.scrollHeight;
+      }, 400);
```

Comment:
Again, this timeout is rather long and arbitrary. Why 400ms?

Reply:
This was taken from the original scroll implementation of the activity tab, but it's removed completely now